targets/TARGET_Cypress/TARGET_PSOC6/psoc6csp/hal/src/cyhal_trng.c

```diff
@@ -36,19 +35,15 @@
 cy_rslt_t cyhal_trng_init(cyhal_trng_t *obj)
 {
     CY_ASSERT(NULL != obj);
-
-    memset(obj, 0, sizeof(cyhal_trng_t));
-    obj->resource.type = CYHAL_RSC_TRNG;
-    return cyhal_crypto_reserve(&(obj->base), &(obj->resource));
+    return cyhal_crypto_reserve(&(obj->base), &(obj->resource), CYHAL_CRYPTO_TRNG);
 }
 
 void cyhal_trng_free(cyhal_trng_t *obj)
 {
-    CY_ASSERT(NULL != obj || obj->resource.type != CYHAL_RSC_TRNG);
+    CY_ASSERT(NULL != obj || obj->resource.type != CYHAL_RSC_CRYPTO);
```

Comment:
Is the TRNG, CRC, and Crypto resource all the same hardware block?

Reply:
Yes, all these three are based on the same Crypto HW block

Reply:
OK